### ISSUE TITLE
Support multiple existing classes as messages

### ISSUE THREAD

**Issue by mkotsbak** [knutwalker/typed-actors#3]
It would be nice if you could reuse existing classes as messages to the typed actors. Works fine if it support only one, but with multiple classes, a union function is needed.

Someone in Gitter suggested using Shapeless coproduct to do this. Actually it works, but is a bit verbose:

``` scala
object OneActor {
    import shapeless._
    type protocol = SomeType :+: Udp.Unbind.type :+: CNil
    val coprod = Coproduct[protocol]
    def props(...) = Props[protocol, OneActor](new OneActor(...))
}
...
actor ! OneActor.coprod(Udp.Unbind)
```

I suspect that it is possible to integrate it into the library so that is is less verbose. What about:

```
def Props[Coproduct[A], T <: Actor : ClassTag](creator: ⇒ T): Props[A] =
that gives a CoproductProps, and Add an ActorOf taking CoproductProps with an "!" packing the argument inside a coproduct (or is an implicit conversion needed here?)
```

**Comment by knutwalker** [comment]
Thanks for the issue! I already have implemented something like this and also though about using `shapeless.Coproduct` at first, but ended up doing a custom union type, see
https://github.com/knutwalker/typed-actors/blob/ca57dfb332f8ba32852df9d633426bcc2058fcca/README.md#union-types and https://github.com/knutwalker/typed-actors/blob/ca57dfb332f8ba32852df9d633426bcc2058fcca/README.md#working-with-union-types or https://github.com/knutwalker/typed-actors/blob/ca57dfb332f8ba32852df9d633426bcc2058fcca/examples/src/test/scala/org/example/UnionExample2.scala for examples. I want to release this as version 1.5 someday this week.

**Comment by knutwalker** [comment]
In the meantime, you can play with the snapshot

```
resolvers +="Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "de.knutwalker" %% "typed-actors" % "1.4.1-SNAPSHOT"
```

**Comment by mkotsbak** [comment]
Ah, great, thanks! Good I didn't try to implement a Coproduct solution.

**Comment by mkotsbak** [comment]
This is solved in 668f40616f2aec6a307f7f547e0a390facf87455 and released in 1.5.0.
